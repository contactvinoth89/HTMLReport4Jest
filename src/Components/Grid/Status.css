.statusicon {
    vertical-align: text-bottom;
}

.passed {
    color: green;
}

.failed {
    color: red;
}
.pending {
    color: orange;
}

.todo {
    color: gray;
}
