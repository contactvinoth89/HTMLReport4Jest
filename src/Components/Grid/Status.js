import React, { Component, Fragment } from 'react';
import './Status.css';
import PropTypes from 'prop-types';
class Status extends Component {
    render() {
        if (this.props.status === 'passed') {
            return (
                <Fragment>
                    <i>
                        <svg
                            aria-label="passed"
                            fill="currentColor"
                            viewBox="0 0 16 16"
                            version="1.1"
                            width="1em"
                            height="1em"
                            role="img"
                            className="passed statusicon"
                        >
                            <path
                                fillRule="evenodd"
                                // eslint-disable-next-line max-len
                                d="M13.78 4.22a.75.75 0 010 1.06l-7.25 7.25a.75.75 0 01-1.06 0L2.22 9.28a.75.75 0 011.06-1.06L6 10.94l6.72-6.72a.75.75 0 011.06 0z"
                            ></path>
                        </svg>
                    </i>
                    {this.props.status}
                </Fragment>
            );
        } else if (this.props.status === 'failed') {
            return (
                <Fragment>
                    <i>
                        <svg
                            aria-label="failed"
                            className="failed statusicon"
                            viewBox="0 0 16 16"
                            version="1.1"
                            width="1em"
                            height="1em"
                            role="img"
                            fill="currentColor"
                        >
                            <path
                                fillRule="evenodd"
                                // eslint-disable-next-line max-len
                                d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"
                            ></path>
                        </svg>
                    </i>
                    {this.props.status}
                </Fragment>
            );
        } else if (this.props.status === 'pending') {
            return (
                <Fragment>
                    <i>
                        <svg
                            className="pending statusicon"
                            viewBox="0 0 16 16"
                            version="1.1"
                            width="1em"
                            height="1em"
                            aria-hidden="true"
                            fill="currentColor"
                        >
                            <path
                                fillRule="evenodd"
                                d="M8 4a4 4 0 100 8 4 4 0 000-8z"
                            ></path>
                        </svg>
                    </i>
                    {this.props.status}
                </Fragment>
            );
        } else if (this.props.status === 'todo') {
            return (
                <Fragment>
                    <i>
                        <svg
                            className="todo statusicon"
                            viewBox="0 0 16 16"
                            version="1.1"
                            width="1em"
                            height="1em"
                            aria-hidden="true"
                            fill="currentColor"
                        >
                            <path
                                fillRule="evenodd"
                                d="M8 4a4 4 0 100 8 4 4 0 000-8z"
                            ></path>
                        </svg>
                    </i>
                    {this.props.status}
                </Fragment>
            );
        } else {
            return <Fragment>{this.props.status}</Fragment>;
        }
    }
}
Status.propTypes = {
    status: PropTypes.string.isRequired,
};
export default Status;
