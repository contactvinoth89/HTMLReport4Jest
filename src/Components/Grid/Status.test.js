import React from 'react';
import { render } from '@testing-library/react';
import Status from './Status';
test('Should contain passed', () => {
    const { container } = render(<Status status="passed" />);
    expect(container.textContent).toEqual('passed');
    expect(container).toMatchSnapshot();
});
test('Should contain failed', () => {
    const { container } = render(<Status status="failed" />);
    expect(container.textContent).toEqual('failed');
    expect(container).toMatchSnapshot();
});
test('Should contain pending', () => {
    const { container } = render(<Status status="pending" />);
    expect(container.textContent).toEqual('pending');
    expect(container).toMatchSnapshot();
});
test('Should contain todo', () => {
    const { container } = render(<Status status="todo" />);
    expect(container.textContent).toEqual('todo');
    expect(container).toMatchSnapshot();
});
