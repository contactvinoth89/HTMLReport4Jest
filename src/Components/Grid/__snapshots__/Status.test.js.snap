// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should contain failed 1`] = `
<div>
  <i>
    <svg
      aria-label="failed"
      class="failed statusicon"
      fill="currentColor"
      height="1em"
      role="img"
      version="1.1"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"
        fill-rule="evenodd"
      />
    </svg>
  </i>
  failed
</div>
`;

exports[`Should contain passed 1`] = `
<div>
  <i>
    <svg
      aria-label="passed"
      class="passed statusicon"
      fill="currentColor"
      height="1em"
      role="img"
      version="1.1"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M13.78 4.22a.75.75 0 010 1.06l-7.25 7.25a.75.75 0 01-1.06 0L2.22 9.28a.75.75 0 011.06-1.06L6 10.94l6.72-6.72a.75.75 0 011.06 0z"
        fill-rule="evenodd"
      />
    </svg>
  </i>
  passed
</div>
`;

exports[`Should contain pending 1`] = `
<div>
  <i>
    <svg
      aria-hidden="true"
      class="pending statusicon"
      fill="currentColor"
      height="1em"
      version="1.1"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M8 4a4 4 0 100 8 4 4 0 000-8z"
        fill-rule="evenodd"
      />
    </svg>
  </i>
  pending
</div>
`;

exports[`Should contain todo 1`] = `
<div>
  <i>
    <svg
      aria-hidden="true"
      class="todo statusicon"
      fill="currentColor"
      height="1em"
      version="1.1"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M8 4a4 4 0 100 8 4 4 0 000-8z"
        fill-rule="evenodd"
      />
    </svg>
  </i>
  todo
</div>
`;
